sum.go

```diff
@@ -52,7 +52,10 @@ func Sum(data []byte, code uint64, length int) (Multihash, error) {
 	default:
 		switch code {
 		case ID:
-			d = sumID(data)
+			d, err = sumID(data, length)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
The `err != nil` check is not needed here as it is checked just after the `switch`.